omero/util_scripts/Move_Annotations.py

```diff
@@ -0,0 +1,185 @@
+    links_dict = {}
+    for l in old_links:
+        ann = l.child
+        links_dict[ann.id.val] = l
+    old_links = links_dict.values()
+
+    new_links = []
+    for l in old_links:
+        print "    Annotation:", l.child.id.val, l.child.__class__.__name__
+        link = WellAnnotationLinkI()
+        link.parent = WellI(well.id, False)
+        link.child = l.child
+        new_links.append(link)
+    try:
+        conn.getUpdateService().saveArray(new_links)
+    except Exception, ex:
+        print "Failed to create links: ", ex.message
+        return 0
+
+    if remove_anns:
+        print "Deleting ImageAnnotation links...", link_ids
+        try:
+            for link_id in link_ids:
+                to_delete = ImageAnnotationLinkI(link_id, False)
+                conn.getUpdateService().deleteObject(to_delete)
+        except Exception, ex:
+            print "Failed to delete links: ", ex.message
+    return len(new_links)
+
+
+def move_annotations(conn, script_params):
+    """Process script parameters and move annotations as specified."""
+    plates = []
+    filter_type = None
+
+    dtype = script_params['Data_Type']
+    ids = script_params['IDs']
+    ann_type = script_params['Annotation_Type']
+    if ann_type in ANN_TYPES:
+        filter_type = ANN_TYPES[ann_type]
+    remove_anns = script_params['Remove_Annotations']
+    ns = script_params.get('Namespace')
+
+    # Get the Plates or Wells
+    objects = conn.getObjects(dtype, ids)
+
+    ann_total = 0
+
+    if dtype == 'Well':
+        for well in objects:
+            ann_count = move_well_annotations(conn, well, filter_type,
+                                              remove_anns, ns)
+            ann_total += ann_count
+    else:
+        if dtype == 'Plate':
+            plates = list(objects)
+        elif dtype == 'Screen':
+            for screen in objects:
+                plates.extend(list(screen.listChildren()))
+        print "Found Plates:", plates
+        for plate in plates:
+            for well in plate.listChildren():
+                ann_count = move_well_annotations(conn, well, filter_type,
+                                                  remove_anns, ns)
+                ann_total += ann_count
+
+    return ann_total
+
+
+def runScript():
+    """The main entry point of the script."""
+    data_types = [rstring('Screen'), rstring('Plate'), rstring('Well')]
+
+    ann_types = [rstring('All')]
+    ann_types.extend([rstring(k) for k in ANN_TYPES.keys()])
+
+    client = scripts.client(
+        'Move_Annotations.py',
+        """Move Annotations from SPW Images to their parent Wells.""",
+
+        scripts.String(
+            "Data_Type", optional=False, grouping="1",
+            description="The data you want to work with.", values=data_types,
+            default="Plate"),
+
+        scripts.List(
+            "IDs", optional=False, grouping="2",
+            description="List of Screen IDs or Plate IDs").ofType(rlong(0)),
```

Comment:
``well`` is supported as a data_type, this need to be reflected in the description of the ids param